### DISCUSSION TITLE
Form Helpers & Tailwind

### DISCUSSION THREAD

**Discussion by mikesklein** [rails/tailwindcss-rails#370] [Q&A]
There is an issue with Rails form helpers and tailwind. Standard fields like text or text area are stylable, but when I try to add styles to form.collection_select, form.select and form.date_select the added classes are getting not applied.

I've never submit and Issue before, so please let me know if you need more information.

**Comment by flavorjones** [comment]
@mikesklein Sorry you're having problems, I'll try to help.

If I use [`collection_select`](https://api.rubyonrails.org/v7.1.3.3/classes/ActionView/Helpers/FormBuilder.html#method-i-collection_select) like this, passing in `html_options` as the final argument:

```
  <div class="my-5">
    <%= form.collection_select(
      :name, ["Sam", "Bob"], :to_s, :to_s,
      {}, {class: "block shadow rounded-md border border-gray-200 outline-none"},
    ) %>
  </div>
```

it renders as

```html
  <div class="my-5">
    <select class="block shadow rounded-md border border-gray-200 outline-none" name="post[name]" id="post_name">
      <option value="Sam">Sam</option>
      <option value="Bob">Bob</option>
    </select>
  </div>
```

and looks fine:

![image](https://github.com/rails/tailwindcss-rails/assets/8207/8fecab78-72da-413e-8a5c-779ecaa16ca3)

Can you provide this level of detail to help us diagnose what's going on?

- Provide the code in your view
- Provide the desired HTML and the actual HTML rendered
